src/Util.php

```diff
@@ -132,4 +132,171 @@ private static function compareArrays(array $valueOne, array $valueTwo) : bool
+     *
+     * @since 2.4.0
+     *
+     * @param int $asType  The type mask to cast to
+     * @param mixed $value The value to cast
+     * @return mixed The cast value
+     */
+    public static function cast(int $asType, $value)
+    {
+        // get the value type
+        $type = self::typeMask($value);
+
+        // check whether value is already one of the desired types
+        if ($type & $asType) {
+            return $value;
+        }
+
+        // cast objects & arrays
+        // -> directly to an object or associative array
+        // -> to a json-encoded string
+        // -> to an integer count of the members
+        // -> to a boolean indicating whether any members are present
+        if ($type & (JsonBrowser::TYPE_OBJECT | JsonBrowser::TYPE_ARRAY)) {
+            if ($asType & JsonBrowser::TYPE_OBJECT) {
+                return (object) $value;
+            } elseif ($asType & JsonBrowser::TYPE_ARRAY) {
+                return (array) $value;
+            } elseif ($asType & JsonBrowser::TYPE_STRING) {
+                return json_encode($value, \JSON_PRETTY_PRINT | \JSON_UNESCAPED_SLASHES | \JSON_UNESCAPED_UNICODE);
+            } elseif ($asType & (JsonBrowser::TYPE_NUMBER | JsonBrowser::TYPE_INTEGER)) {
+                return count((array)$value);
+            } elseif ($asType & JsonBrowser::TYPE_BOOLEAN) {
+                return (bool) count((array)$value);
+            }
+        }
+
+        // cast strings
+        // -> to an array of unicode characters
+        // -> to an array of unicode characters, cast as an object (stdClass)
+        // -> to an integer count of [unicode] characters in the string
+        // -> to a boolean indicating whether the string length is greater than zero
+        if ($type & JsonBrowser::TYPE_STRING) {
+            if ($asType & JsonBrowser::TYPE_ARRAY) {
+                return preg_split('//u', $value, null, \PREG_SPLIT_NO_EMPTY);
+            } elseif ($asType & JsonBrowser::TYPE_OBJECT) {
+                return (object) preg_split('//u', $value, null, \PREG_SPLIT_NO_EMPTY);
+            } elseif ($asType & (JsonBrowser::TYPE_NUMBER | JsonBrowser::TYPE_INTEGER)) {
+                return mb_strlen($value);
+            } elseif ($asType & JsonBrowser::TYPE_BOOLEAN) {
+                return (bool) strlen($value);
+            }
+        }
+
+        // cast numbers
+        // -> to an integer (or float if larger than PHP_INT_MAX), with the fractional component discarded
+        // -> to a string representation of the number, in base-10
+        // -> to the only member [0] of an array
+        // -> to the 'value' property of an stdClass object
+        // -> to a boolean indicating whether the number is non-zero
+        if ($type & JsonBrowser::TYPE_NUMBER) {
+            if ($asType & JsonBrowser::TYPE_INTEGER) {
+                return $value > \PHP_INT_MAX ? floor($value) : (int) floor($value);
```

Comment:
> $value > \PHP_INT_MAX

Not sure that makes sense; if a number is greater than PHP_INT_MAX then how can we do numerical checks on it?

Reply:
Because floats can be larger than `PHP_INT_MAX`. So in cases where a value larger than `PHP_INT_MAX` is present, it will still return an integer (via `floor()`), but in a float type.

Reply:
Hmmm, a thought though - large floats can lose precision. Maybe it's better to just cast as int regardless, and let the user worry about cases where it's too large. Thoughts?